vyper/semantics/types/indexable/sequence.py

```diff
@@ -152,6 +152,11 @@ def __init__(
         is_public: bool = False,
         is_immutable: bool = False,
     ) -> None:
+        # TODO fix circular import
+        from vyper.semantics.types.indexable.mapping import MappingDefinition
+
+        if isinstance(value_type, MappingDefinition):
```

Comment:
i think you can use `value_type._as_array`

Reply:
I am encountering an issue where the `value_type` is a `BaseTypeDefinition` and therefore lacking the `_as_array` property. Should we propagate `_as_array` to the `BaseTypeDefinition`?

Reply:
yea as_array should be a property on all types. you could always use `getattr(value_type, "_as_array", False)`. but hold off on this, #2974 will introduce a new way of doing things

Reply:
With the new type system, `_as_array` is still an issue for dynamic arrays of strings and byte arrays. This is due to a difference in syntax between static and dynamic arrays, where the latter allows strings and byte arrays as value type but not the former.

Reply:
the bytestring override should have a comment explaining why there is a carveout/ asymmetry, and that it will be changed in future versions of vyper once https://github.com/vyperlang/vyper/issues/2733 is implemented